pkg/search/postgres/common.go

```diff
@@ -169,19 +184,31 @@ func (q *query) getPortionBeforeFromClause() string {
 		return fmt.Sprintf("select %q.serialized", q.From)
 	case SEARCH:
 		// Always select the primary keys first.
-		var selectPaths []string
-		for _, field := range q.PrimaryKeyFields {
-			selectPaths = append(selectPaths, field.PathForSelectPortion())
-		}
-		for _, field := range q.SelectedFields {
-			selectPaths = append(selectPaths, field.PathForSelectPortion())
+		allSelectFields := q.PrimaryKeyFields
+		allSelectFields = append(allSelectFields, q.SelectedFields...)
+		allSelectFields = append(allSelectFields, q.ExtraSelectedFieldPaths()...)
+
+		if !q.shouldAggrRows() {
+			selectStrs := make([]string, 0, len(allSelectFields))
+			for _, f := range allSelectFields {
+				selectStrs = append(selectStrs, f.PathForSelectPortion())
+			}
+			return "select " + stringutils.JoinNonEmpty(", ", selectStrs...)
 		}
 
-		// Extra fields are group by fields and order by fields.
-		for _, field := range q.ExtraSelectedFieldPaths() {
-			selectPaths = append(selectPaths, field.PathForSelectPortion())
+		// Collect aggregated and non-aggregated fields separately because aggregate functions cannot be nested.
+		var nonAggrSelects, aggrSelects []string
+		for _, field := range allSelectFields {
+			if field.AggrAndUnnest() {
+				// Alias don't work in SQL `row()` function.
+				field.Alias = ""
+				aggrSelects = append(aggrSelects, field.PathForSelectPortion())
+			} else {
+				nonAggrSelects = append(nonAggrSelects, field.PathForSelectPortion())
+			}
 		}
-		return "select " + stringutils.JoinNonEmpty(", ", selectPaths...)
+		return "select jsonb_agg(row(" + stringutils.JoinNonEmpty(", ", aggrSelects...) + ")), " +
```

Comment:
`row` result will be converted to `jsonb` using `to_json` function, resulting in field names like "f1","f2", etc. Maybe it would be beneficial to construct the jsonb object directly? In this way one could pass column names for keys, and in the end:

```
select jsonb_build_object('id', 1, 'name', 'test');
    jsonb_build_object
---------------------------
 {"id": 1, "name": "test"}
```

vs

```
select to_json(row(1, 'test'));
       to_json
----------------------
 {"f1":1,"f2":"test"}
```

Reply:
Yes, I looked at that option as well. I didn't want the query string to be too long hence went with the current approach. Additionally, we do not really use the select path names (except for debugging). 

I will try it but might do it as a polish task in one of the follow-ups.